src/coordination/coordinator_log_store.cpp

```diff
@@ -13,229 +13,164 @@
-    (*ret)[cc++] = MakeClone(src);
+    logs_[index] = clone;
   }
-  return ret;
 }
 
-// NOLINTNEXTLINE(google-default-arguments)
-ptr<std::vector<ptr<log_entry>>> CoordinatorLogStore::log_entries_ext(ulong start, ulong end,
-                                                                      int64 batch_size_hint_in_bytes) {
-  ptr<std::vector<ptr<log_entry>>> ret = cs_new<std::vector<ptr<log_entry>>>();
-
-  if (batch_size_hint_in_bytes < 0) {
-    return ret;
-  }
+ptr<std::vector<ptr<log_entry>>> CoordinatorLogStore::log_entries(uint64_t start, uint64_t end) {
+  auto ret = cs_new<std::vector<ptr<log_entry>>>();
+  ret->resize(end - start);
 
-  size_t accum_size = 0;
-  for (ulong ii = start; ii < end; ++ii) {
+  for (uint64_t i = start, curr_index = 0; i < end; ++i, ++curr_index) {
     ptr<log_entry> src = nullptr;
     {
-      std::lock_guard<std::mutex> l(logs_lock_);
-      auto entry = logs_.find(ii);
-      if (entry == logs_.end()) {
-        entry = logs_.find(0);
-        assert(0);
+      auto lock = std::lock_guard{logs_lock_};
+      if (auto const entry = logs_.find(i); entry != logs_.end()) {
+        src = entry->second;
+      } else {
+        throw RaftCouldNotFindEntryException("Could not find entry at index {}", i);
       }
-      src = entry->second;
     }
-    ret->push_back(MakeClone(src));
-    accum_size += src->get_buf().size();
-    if (batch_size_hint_in_bytes && accum_size >= (ulong)batch_size_hint_in_bytes) break;
+    (*ret)[curr_index] = MakeClone(src);
   }
   return ret;
 }
 
-ptr<log_entry> CoordinatorLogStore::entry_at(ulong index) {
+ptr<log_entry> CoordinatorLogStore::entry_at(uint64_t index) {
   ptr<log_entry> src = nullptr;
   {
-    std::lock_guard<std::mutex> l(logs_lock_);
-    auto entry = logs_.find(index);
-    if (entry == logs_.end()) {
-      entry = logs_.find(0);
-    }
-    src = entry->second;
+    auto lock = std::lock_guard{logs_lock_};
+    src = FindOrDefault_(index);
   }
   return MakeClone(src);
 }
 
-ulong CoordinatorLogStore::term_at(ulong index) {
-  ulong term = 0;
+uint64_t CoordinatorLogStore::term_at(uint64_t index) {
+  uint64_t term = 0;
   {
-    std::lock_guard<std::mutex> l(logs_lock_);
-    auto entry = logs_.find(index);
-    if (entry == logs_.end()) {
-      entry = logs_.find(0);
-    }
-    term = entry->second->get_term();
+    auto lock = std::lock_guard{logs_lock_};
+    term = FindOrDefault_(index)->get_term();
   }
   return term;
 }
 
-ptr<buffer> CoordinatorLogStore::pack(ulong index, int32 cnt) {
+ptr<buffer> CoordinatorLogStore::pack(uint64_t index, int32 cnt) {
   std::vector<ptr<buffer>> logs;
 
   size_t size_total = 0;
-  for (ulong ii = index; ii < index + cnt; ++ii) {
+  uint64_t const end_index = index + cnt;
+  for (uint64_t i = index; i < end_index; ++i) {
     ptr<log_entry> le = nullptr;
     {
-      std::lock_guard<std::mutex> l(logs_lock_);
-      le = logs_[ii];
+      auto lock = std::lock_guard{logs_lock_};
+      le = logs_[i];
```

Comment:
Similar to above. Not sure how the ptr<> works, is it safe to handle it without holding the lock?

Reply:
Should be, this part is used from the example. Whole log_store is, I just stripped parts we didn't need.